src/box/lua/config/init.lua

```diff
@@ -450,6 +453,9 @@ function methods._reload_noexc(self, opts)
         self:_store(self:_collect(opts))
         self:_apply()
         self:_set_status_based_on_alerts()
+        if self._status == 'ready' or self._status == 'check_warnings' then
+            self._active_metadata = self._metadata
+        end
```

Comment:
I think we should do it after the post_apply phase (after roles and the app execution, but before extras.post_apply). This way if a role/app execution failed we don't lie about active configuration.

In fact, this situation means some intermediate active configuration, but I would expect that `active` will show the latest one that was fully and successfuly applied.

Please, hold this property with a test: a fail in role/app applying should lead to different `active` and `last` metadata.

Reply:
I am not sure about this. I mean, this way the `active` meta in roles and app will be from the last successful load, but `configdata` will be from the current load.

Reply:
We do the same for statuses.

There is an alternative: we can set the active metadata after `apply`, but revert it back if an error occurs in `post_apply`.

However, I think that it would be too complicated, while it is not necessary that it was expected by a user. I would prefer a simple solution.

Since we provide both `last` and `active` I don't think that it worth to do the tricks like we do for `config:get()`.

Reply:
Agree. Fixed.